src/rnnlm/arpa-sampling.h

```diff
@@ -0,0 +1,182 @@
+// arpa_sampling.h
+
+// Copyright     2016  Ke Li
+
+// See ../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ARPA_SAMPLING_H_
+#define ARPA_SAMPLING_H_
+
+#include <sys/time.h>
+#include <unistd.h>
+#include "lm/arpa-file-parser.h"
+#include "fst/fstlib.h"
+
+#ifdef _MSC_VER
```

Comment:
I think I mentioned there is no need to write these #ifdef things for undered_map.

Check other files in kaldi that uses unordered_map to see how it's done